<a href="https://colab.research.google.com/github/WaadK98/Amazon_content_from_URL/blob/master/EDA__Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import requests

UCI Data Repository

1- Heart Disease Data Set: https://archive.ics.uci.edu/ml/datasets/Heart+Disease                                                                  2- Forest Fires Data Set: https://archive.ics.uci.edu/ml/datasets/Forest+Fires
3- Travel Reviews Data Set: https://archive.ics.uci.edu/ml/datasets/Travel+Reviews

In [ ]:
heart= pd.read_csv("/Users/waadalsaleh/Desktop/Data_BootCamp/Stack_3/heart.csv")
forest=pd.read_csv("/Users/waadalsaleh/Desktop/Data_BootCamp/Stack_3/forestfires.csv")
travel= pd.read_csv("/Users/waadalsaleh/Desktop/Data_BootCamp/Stack_3/tripadvisor_review.csv")

In [ ]:
heart

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
0        0   0     1       1  
1        0   0     2       1  
2        2   0     2       1  
3        2   0     2       1  
4        2   0     2       1  
..     ...  ..   ...     ...  
298      1   0     3       0  
299      1   0     3       0  
300      1   2     3       0  
301      1   1     3       0  
302      1   1     2       0  

[303 rows x 14 columns]

In [ ]:
forest

X  Y month  day  FFMC    DMC     DC   ISI  temp  RH  wind  rain   area
0    7  5   mar  fri  86.2   26.2   94.3   5.1   8.2  51   6.7   0.0   0.00
1    7  4   oct  tue  90.6   35.4  669.1   6.7  18.0  33   0.9   0.0   0.00
2    7  4   oct  sat  90.6   43.7  686.9   6.7  14.6  33   1.3   0.0   0.00
3    8  6   mar  fri  91.7   33.3   77.5   9.0   8.3  97   4.0   0.2   0.00
4    8  6   mar  sun  89.3   51.3  102.2   9.6  11.4  99   1.8   0.0   0.00
..  .. ..   ...  ...   ...    ...    ...   ...   ...  ..   ...   ...    ...
512  4  3   aug  sun  81.6   56.7  665.6   1.9  27.8  32   2.7   0.0   6.44
513  2  4   aug  sun  81.6   56.7  665.6   1.9  21.9  71   5.8   0.0  54.29
514  7  4   aug  sun  81.6   56.7  665.6   1.9  21.2  70   6.7   0.0  11.16
515  1  4   aug  sat  94.4  146.0  614.7  11.3  25.6  42   4.0   0.0   0.00
516  6  3   nov  tue  79.5    3.0  106.7   1.1  11.8  31   4.5   0.0   0.00

[517 rows x 13 columns]

In [ ]:
travel

User ID  Category 1  Category 2  Category 3  Category 4  Category 5  \
0      User 1        0.93        1.80        2.29        0.62        0.80   
1      User 2        1.02        2.20        2.66        0.64        1.42   
2      User 3        1.22        0.80        0.54        0.53        0.24   
3      User 4        0.45        1.80        0.29        0.57        0.46   
4      User 5        0.51        1.20        1.18        0.57        1.54   
..        ...         ...         ...         ...         ...         ...   
975  User 976        0.74        1.12        0.30        0.53        0.88   
976  User 977        1.25        0.92        1.12        0.38        0.78   
977  User 978        0.61        1.32        0.67        0.43        1.30   
978  User 979        0.93        0.20        0.13        0.43        0.30   
979  User 980        0.93        0.56        1.13        0.51        1.34   

     Category 6  Category 7  Category 8  Category 9  Category 10  
0          2.42        3.19        2.79        1.82         2.42  
1          3.18        3.21        2.63        1.86         2.32  
2          1.54        3.18        2.80        1.31         2.50  
3          1.52        3.18        2.96        1.57         2.86  
4          2.02        3.18        2.78        1.18         2.54  
..          ...         ...         ...         ...          ...  
975        1.38        3.17        2.78        0.99         3.20  
976        1.68        3.18        2.79        1.34         2.80  
977        1.78        3.17        2.81        1.34         3.02  
978        0.40        3.18        2.98        1.12         2.46  
979        2.36        3.18        2.87        1.34         2.40  

[980 rows x 11 columns]

The Heart Disease Data Set originaly has 76 attributes but the majority of the published expermants only use these 14 attributes:
1. age: age in years
2. sex: 1=male, 1= female
3. cp: chest pain type, Value 1: typical angina, Value 2: atypical angina, Value 3: non-anginal pain, Value 4: asymptomatic
4. trestbps: resting blood pressure in mm
5. chol: serum cholestoral in mg/dl
6. fbs: fasting blood suger >120 (1=true) (0=false)
7. restecg: resting electrocardiographic results, Value 0: normal, Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV), Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria
8. thalach: maximum heart rate achieved
9. exang: exercise induced angina (1 = yes; 0 = no)
10. oldpeak: ST depression induced by exercise relative to rest
11. slope: the slope of the peak exercise ST segment, Value 1: upsloping, Value 2: flat, Value 3: downsloping
12. ca: number of major vessels (0-3) colored by flourosopy
13. thal: 3 = normal; 6 = fixed defect; 7 = reversable defect
14. num(the predicted attribute): diagnosis of heart disease (angiographic disease status), Value 0: < 50% diameter narrowing, Value 1: > 50% diameter narrowing
    
    

OpenAQ Saudi Arabia data

In [ ]:
#API link from OpenAQ
PATH_API = 'https://u50g7n0cbj.execute-api.us-east-1.amazonaws.com/v2/cities?limit=100&page=1&offset=0&sort=asc&country_id=SA&order_by=city'

r = requests.get(PATH_API)
data = r.json()

In [ ]:
data

{'meta': {'name': 'openaq-api',
  'license': 'CC BY 4.0d',
  'website': 'https://u50g7n0cbj.execute-api.us-east-1.amazonaws.com/',
  'page': 1,
  'limit': 100,
  'found': 2},
 'results': [{'country': 'SA',
   'city': 'Dharan',
   'count': 24,
   'locations': 1,
   'firstUpdated': '2019-07-15T19:00:00+00:00',
   'lastUpdated': '2019-07-16T18:00:00+00:00',
   'parameters': ['pm25']},
  {'country': 'SA',
   'city': 'N/A',
   'count': 3822233,
   'locations': 2,
   'firstUpdated': '2019-10-28T18:00:00+00:00',
   'lastUpdated': '2021-01-20T18:00:00+00:00',
   'parameters': ['pm25']}]}

1. What cities are in the data?
There is only on city in this data which is 'Dharan' the other one is 'N/A'


In [ ]:
df = pd.DataFrame(list(data.items())) 

In [ ]:
df.shape

(2, 2)

In [ ]:
df

0                                                  1
0     meta  {'name': 'openaq-api', 'license': 'CC BY 4.0d'...
1  results  [{'country': 'SA', 'city': 'Dharan', 'count': ...

2. How many rows and columns are in the data from your API call
there are 2 columns and 2 rows 